File: synthetic_test/syn.go
package main

import (
	"os"
	"strconv"
)

func main() {
	iters_str := os.Getenv("SPEEDTEST_ITERS")
	iters, _ := strconv.Atoi(iters_str)

	for i := 0; i < iters; i++ {
		var array [1000]int

		for j := 0; j < 100; j++ { array[j] = j }

		for j := 0; j < 10; j++ {
			tmp := array[j]
			array[j] = array[1000 - (10 - j)]
			array[1000 - (10 - j)] = tmp
		}
	}
}

File: synthetic_test/syn.lua
local os = require("os")

local iters = tonumber(os.getenv("SPEEDTEST_ITERS"))
for _ = 1, iters do
    local a = {}
    for j = 1, 1000 do
        a[j] = 0
    end

    for j = 1, 100 do
        a[j] = j - 1
    end

    for j = 1, 10 do
        local tmp = a[j]
        a[j] = a[1000 - (10 - j)]
        a[1000 - (10 - j)] = tmp
    end
end
